packages/loader/driver-utils/src/runWithRetry.ts

```diff
@@ -107,8 +107,10 @@ export async function runWithRetry<T>(
 			numRetries++;
 			lastError = err;
 			// If the error is throttling error, then wait for the specified time before retrying.
-			// If the waitTime is not specified, then we start with retrying immediately to max of 8s.
-			retryAfterMs = getRetryDelayFromError(err) ?? Math.min(retryAfterMs * 2, 8000);
+			// If the waitTime is not specified, then we start with retrying immediately to max of 8s or 30s.
+			retryAfterMs =
+				getRetryDelayFromError(err) ??
+				Math.min(retryAfterMs * 2, calculateMaxWaitTime(lastError));
```

Comment:
while lastError & err are the same thing here, it would be better to use same variable on lines 112 & 113.